packages/nemo-evaluator/src/nemo_evaluator/client/client.py

```diff
@@ -0,0 +1,356 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""NeMo Evaluator client with integrated adapter support for client-mode evaluation."""
+
+import asyncio
+import os
+from typing import Any, List, Optional
+
+from openai import AsyncOpenAI
+from tenacity import (
+    AsyncRetrying,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential,
+)
+from tqdm.asyncio import tqdm_asyncio
+
+from nemo_evaluator.api.api_dataclasses import EndpointModelConfig
+from nemo_evaluator.client.adapter_transport import create_async_adapter_http_client
+from nemo_evaluator.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class NeMoEvaluatorClient:
+    """Async OpenAI-compatible client with integrated adapter pipeline support.
+
+    This client enables "client mode" for adapters, where the adapter interceptor
+    pipeline runs in-process through the HTTP client, instead of requiring a
+    separate proxy server.
+
+    Args:
+        endpoint_model_config: Endpoint model configuration
+        output_dir: Directory for adapter output files
+    """
+
+    def __init__(
+        self,
+        endpoint_model_config: EndpointModelConfig,
+        output_dir: str,
+    ):
+        self.model_id = endpoint_model_config.model_id
+        self.model_url = endpoint_model_config.url
+        self.api_key = endpoint_model_config.api_key
+        self.adapter_config = endpoint_model_config.adapter_config
+        self.temperature = endpoint_model_config.temperature
+        self.top_p = endpoint_model_config.top_p
+        self.max_new_tokens = endpoint_model_config.max_new_tokens
+        self.request_timeout = endpoint_model_config.request_timeout
+        self.max_retries = endpoint_model_config.max_retries or 5
+        self.parallelism = endpoint_model_config.parallelism or 1
+        self.output_dir = output_dir
+
+        is_base_url = endpoint_model_config.is_base_url or False
+
+        adapter_http_client, self.adapter_transport = create_async_adapter_http_client(
+            endpoint_url=self.model_url,
+            adapter_config=self.adapter_config,
+            output_dir=self.output_dir,
+            is_base_url=is_base_url,
+            model_name=self.model_id,
+        )
+
+        self.client = AsyncOpenAI(
+            http_client=adapter_http_client,
+            base_url=self.model_url,
+            api_key=os.getenv(self.api_key, "dummy_api_key")
```

Comment:
I think we should fail if `api_key` env is not defined instead of using a default here